### PR TITLE
sane remote logging/stderr, fixes #461

### LINKED ISSUE #461: remote logging configuration / output selection
Noticed that the `borg check repo` progress indicator does not show with remote repos.

Seems like there are multiple issues:
- flush() issue?
- logging level of remote borg in sync with local borg?
- if we use BORG_LOGGING_CONF=logging.conf, how do we handle the remote borg?
- in general (maybe not related to THIS issue): topical flags (like --progress) given to remote borg?

***

:moneybag: [there is a bounty for this](https://www.bountysource.com/issues/28836113-remote-logging-configuration-output-selection)

### PR DESCRIPTION
**Pull request by ThomasWaldmann**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -121,8 +121,7 @@ Links
  * `GitHub <https://github.com/borgbackup/borg>`_
  * `Issue Tracker <https://github.com/borgbackup/borg/issues>`_
  * `Bounties & Fundraisers <https://www.bountysource.com/teams/borgbackup>`_
- * `New Mailing List <https://mail.python.org/mailman/listinfo/borgbackup>`_
- * `(Old Mailing List's Archives <http://librelist.com/browser/borgbackup/>`_)
+ * `Mailing List <https://mail.python.org/mailman/listinfo/borgbackup>`_
```
Comment (anarcat) [inline README.rst]: this change seems unrelated

**Comment by anarcat** [comment]
in fact, i don't know where to start to review this... this seems like a lot of unrelated changes...

**Comment by ThomasWaldmann** [comment]
maybe read the single commits, that's better than the full diff.

**Comment by anarcat** [comment]
well that's exactly what i mean: it would be easier to review this if the topics would be grouped in topical merge requests instead of bunched together like this.

**Comment by ThomasWaldmann** [comment]
all except the first that crept in here somehow, they are all related, so i won't split them up.
the 2nd and 3rd are just preparating refactorings for the 4th.
